homeassistant/components/lg_thinq/sensor.py

```diff
@@ -0,0 +1,315 @@
+"""Support for sensor entities."""
+
+from __future__ import annotations
+
+import logging
+
+from thinqconnect import DeviceType
+from thinqconnect.devices.const import Property as ThinQProperty
+from thinqconnect.integration import ThinQPropertyEx
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+    PERCENTAGE,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ThinqConfigEntry
+from .coordinator import DeviceDataUpdateCoordinator
+from .entity import ThinQEntity
+
+ERROR_DESC = SensorEntityDescription(
+    key=ThinQPropertyEx.ERROR,
+    translation_key=ThinQPropertyEx.ERROR,
+)
+
+AIR_QUALITY_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.PM1: SensorEntityDescription(
+        key=ThinQProperty.PM1,
+        device_class=SensorDeviceClass.PM1,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    ThinQProperty.PM2: SensorEntityDescription(
+        key=ThinQProperty.PM2,
+        device_class=SensorDeviceClass.PM25,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    ThinQProperty.PM10: SensorEntityDescription(
+        key=ThinQProperty.PM10,
+        device_class=SensorDeviceClass.PM10,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    ThinQProperty.HUMIDITY: SensorEntityDescription(
+        key=ThinQProperty.HUMIDITY,
+        device_class=SensorDeviceClass.HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    ThinQProperty.MONITORING_ENABLED: SensorEntityDescription(
+        key=ThinQProperty.MONITORING_ENABLED,
+        translation_key=ThinQProperty.MONITORING_ENABLED,
+    ),
```

Comment:
Would it make sense to extend `SensorEntityDescription` to have a value_fn?

```suggestion
    ThinQProperty.MONITORING_ENABLED: SensorEntityDescription(
        key=ThinQProperty.MONITORING_ENABLED,
        translation_key=ThinQProperty.MONITORING_ENABLED,
        value_fn=lambda value: value.lower(),
        options=["on_working", "always"],
    ),
```

This would transform it to `on_working` and `always`. For the number enums you can do
```py
ODOR_MAPPING = {
    0: "bad",
}
```

```py
    ThinQProperty.MONITORING_ENABLED: SensorEntityDescription(
        key=ThinQProperty.MONITORING_ENABLED,
        translation_key=ThinQProperty.MONITORING_ENABLED,
        value_fn=lambda value: ODOR_MAPPING.get(value),
        options=list(ODOR_MAPPING.values()),
    ),
```

How does this sound?

Reply:
We used the translation with string.json.